### PR TITLE
[#1237] fix(trino): Fix dropping iceberg schema failure

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Instead of catching exceptions when dropping schemas fails, throw them. 

### Why are the changes needed?

We can't determine whether the schema drop was successful or not based on its return value. 

Fix: #1237 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Existing UTs and ITs.

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorMetadata.java [hunk 1]
```diff
@@ -151,8 +150,11 @@ public void dropSchema(String schemaName, boolean cascade) {
     try {
       schemaCatalog.dropSchema(
           NameIdentifier.ofSchema(metalake.name(), catalogName, schemaName), cascade);
-    } catch (NonEmptySchemaException e) {
-      throw new TrinoException(GRAVITINO_SCHEMA_NOT_EMPTY, "Schema does not empty", e);
+    } catch (Exception e) {
+      String errorMessage =
```
[the remark this answers is not in the document]
Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorMetadata.java:154]: @Clearvive ,
 I can't get your meaning clearly, could you provide more details about it?

[the remark this answers is not in the document]
Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorMetadata.java:154]: This is a common practice, and I'll refine it later. I have created the issue #1249

**Comment by jerryshao** [comment]
@Clearvive can you please help to review.

File: api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java [hunk 2]
```diff
@@ -94,20 +94,22 @@ Schema createSchema(NameIdentifier ident, String comment, Map<String, String> pr
    * @param changes The metadata changes to apply.
    * @return The altered schema.
    * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws UnsupportedOperationException If the catalog does not support altering a schema.
    */
-  Schema alterSchema(NameIdentifier ident, SchemaChange... changes) throws NoSuchSchemaException;
+  Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException, UnsupportedOperationException;
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java:100]: We typically use `IllegalArgumentException` for unsupported changes. please see:

```java
  /**
   * Apply the {@link TableChange change} to a table in the catalog.
   *
   * <p>Implementations may reject the change. If any change is rejected, no changes should be
   * applied to the table.
   *
   * @param ident A table identifier.
   * @param changes Table changes to apply to the table.
   * @return The updated table metadata.
   * @throws NoSuchTableException If the table does not exist.
   * @throws IllegalArgumentException If the change is rejected by the implementation.
   */
  Table alterTable(NameIdentifier ident, TableChange... changes)
      throws NoSuchTableException, IllegalArgumentException;
```

I think we should unify them.

File: api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java [hunk 3]
```diff
@@ -94,20 +94,22 @@ Schema createSchema(NameIdentifier ident, String comment, Map<String, String> pr
...
    *
-   * <p>If the catalog implementation does not support this operation, it may throw {@link
-   * UnsupportedOperationException}.
-   *
    * @param ident The name identifier of the schema.
    * @param cascade If true, recursively drop all objects within the schema.
    * @return True if the schema exists and is dropped successfully, false otherwise.
    * @throws NonEmptySchemaException If the schema is not empty and cascade is false.
+   * @throws UnsupportedOperationException If the catalog does not support dropping a schema with a
+   *     cascade option.
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java:111]: What's the purpose of explicit defining of `UnsupportedOperationException`?

File: api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java (hunk 2 above)
Comment (yuqi1129) [reply api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java:100]: Is `UnsupportedOperationException` more clear and detailed than `IllegalArgumentException`?

File: api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java (hunk 3 above)
Comment (yuqi1129) [reply api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java:111]: I think we should provide detailed error messages if some operations are not supported by certain catalogs.

File: api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java (hunk 2 above)
Comment (yuqi1129) [reply api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java:100]: OK,  I will undo this change here.

**Comment by yuqi1129** [comment]
@jerryshao 

All unnecessary changes have been undo,  please take a look again.

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java [hunk 4]
```diff
@@ -379,24 +379,15 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
   @Override
   public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
```
Comment (diqiu50) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:380]: Will it only throw the NonEmptySchemaException?

**Review by diqiu50 — COMMENTED** [review]
LGTM

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/trino/TrinoConnectorIT.java [hunk 5]
```diff
@@ -842,6 +843,43 @@ void testIcebergCatalogCreatedByGravitino() throws InterruptedException {
...
+            "Created by gravitino client",
+            ImmutableMap.<String, String>builder().build());
+
+    sql =
+        String.format("show schemas in \"%s.%s\" like '%s'", metalakeName, catalogName, schemaName);
+    Assertions.assertTrue(checkTrinoHasLoaded(sql, 30));
+
+    final String sql1 =
+        String.format("drop schema \"%s.%s\".%s cascade", metalakeName, catalogName, schemaName);
+    // Will failed because the iceberg catalog does not support cascade drop
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/trino/TrinoConnectorIT.java:860]: -> Will fail

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java (hunk 4 above)
Comment (yuqi1129) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:380]: Other exceptions will also be thrown, but it's not explicitly declared.

**Comment by yuqi1129** [comment]
@jerryshao 

API related changes have been reverted.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**